Assets/_BForBoss/_Core/Scripts/Boss/DerekContextManager.cs

```diff
@@ -0,0 +1,175 @@
+            
+            if (debug_currentPhase == Phase.Death)
+            {
+                Debug_Reset();
+            }
+        }
+
+        [Button]
+        private void Debug_SwitchVulnerability()
+        {
+            if (debug_rlm == null || debug_dbm == null)
+            {
+                Debug_Initialize();
+            }
+
+            switch (debug_currentVulnerability)
+            {
+                case Vulnerability.Invulnerable:
+                    debug_currentVulnerability = Vulnerability.Vulnerable;
+                    break;
+                case Vulnerability.Vulnerable:
+                    debug_currentVulnerability = Vulnerability.Invulnerable;
+                    break;
+            }
+            
+            debug_dbm.UpdateVulnerability(debug_currentVulnerability);
+        }
+#endif
+        
+        
+        public void Reset()
+        {
+            if (_ringLaborManager == null || _bossManager == null)
+            {
+                return;
+            }
+            
+            _ringLaborManager.Reset();
+            _bossManager.Reset();
+            
+            //TutorialEndButton.Bind(this, BeginEncounter); 
+        }
+
+        public void Initialize(RingLaborManager ringLaborManager, DerekBossManager bossManager)
+        {
+            _ringLaborManager = ringLaborManager;
+            _bossManager = bossManager;
+            
+            //TutorialEndButton.Bind(this, OnPhaseChanged); 
+            _LaborIncompleteWaitForSeconds = new WaitForSeconds(_LaborIncompleteTimer);
+        }
+        
+        private void OnLaborCompleted(bool wasSuccessful)
+        {
+            if (_currentPhase == Phase.Tutorial || _currentVulnerability != Vulnerability.Invulnerable)
+            {
+                Perigon.Utility.Logger.LogError("Labor should not have have been active during the tutorial phase, or when the boss is not invulnerable. Exiting Encounter");
+                return;
+            }
+
+            if (wasSuccessful)
+            {
+                _bossManager.UpdateVulnerability(Vulnerability.Vulnerable);
+            }
+            else
+            {
+                StartCoroutine(ResetLaborTimer());
+            }
+        }
+
+        private void OnPhaseChanged()
+        {
+            switch (_currentPhase)
+            {
+                case Phase.Tutorial:
+                    //TutorialEndButton.UnBind();
+                    _currentPhase = Phase.FirstPhase;
+                    break;
+                case Phase.FirstPhase:
+                    _currentPhase = Phase.SecondPhase;
+                    break;
+                case Phase.SecondPhase:
+                    _currentPhase = Phase.FinalPhase;
+                    break;
+                case Phase.Death:
+                    return;
+                default:
+                    break;
+            }
+
+            _ringLaborManager.Initialize(OnLaborCompleted);
```

Comment:
The method name `Initialize` implies that you should only be calling this method once in the beginning. But it's placed within a method that is called multiple times?

Reply:
Seconding this, I think you should be using `Reset` here and pulling your `OnLaborCompleted` out into its own method. `RingLaborManager.CreateSystems` is really only intended to be called once